File: artifacts/caption-cloner/scripts/test-local-whisper.mjs
/**
 * Path 1 smoke test: Whisper tiny.en in-process, no xAI key.
 * Transcribes the public JFK sample and prints word timestamps.
 */
import { pipeline, env } from "@huggingface/transformers";

env.allowLocalModels = false;

const SAMPLE =
  "https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/jfk.wav";

const started = Date.now();
console.log("[path1] loading Xenova/whisper-tiny.en …");

const asr = await pipeline("automatic-speech-recognition", "Xenova/whisper-tiny.en", {
  dtype: "q8",
});

console.log(`[path1] model ready in ${((Date.now() - started) / 1000).toFixed(1)}s`);
console.log("[path1] transcribing JFK sample …");

const output = await asr(SAMPLE, {
  language: "english",
  task: "transcribe",
  return_timestamps: "word",
});

const words = Array.isArray(output.chunks)
  ? output.chunks.map((c) => ({
      text: String(c.text || "").trim(),
      start: Array.isArray(c.timestamp) ? c.timestamp[0] : 0,
      end: Array.isArray(c.timestamp) ? c.timestamp[1] : 0,
    }))
  : [];

console.log("[path1] text:", String(output.text || "").trim());
console.log("[path1] words:", words.length);
for (const w of words.slice(0, 24)) {
  console.log(`  ${Number(w.start).toFixed(2)}–${Number(w.end).toFixed(2)}  ${w.text}`);
}

if (!String(output.text || "").toLowerCase().includes("ask")) {
  console.error("[path1] FAIL: expected JFK line to contain 'ask'");
  process.exit(1);
}
if (words.length < 5) {
  console.error("[path1] FAIL: expected word timestamps");
  process.exit(1);
}

console.log(`[path1] PASS in ${((Date.now() - started) / 1000).toFixed(1)}s — no xAI used`);

File: artifacts/caption-cloner/scripts/test-local-whisper.py
#!/usr/bin/env python3
"""Path 1 smoke test: local Whisper, no xAI API."""
from __future__ import annotations

import sys
import time
import urllib.request
from pathlib import Path

SAMPLE_URL = "https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/jfk.wav"
CACHE = Path("/tmp/caption-cloner-path1")
CACHE.mkdir(parents=True, exist_ok=True)
SAMPLE = CACHE / "jfk.wav"

started = time.time()
print("[path1] local Whisper test — no XAI_API_KEY")

if not SAMPLE.exists():
    print("[path1] downloading JFK sample…")
    urllib.request.urlretrieve(SAMPLE_URL, SAMPLE)
print(f"[path1] sample: {SAMPLE} ({SAMPLE.stat().st_size} bytes)")

from faster_whisper import WhisperModel

print("[path1] loading faster-whisper tiny.en (cpu, int8)…")
model = WhisperModel("tiny.en", device="cpu", compute_type="int8")
print(f"[path1] model ready in {time.time() - started:.1f}s")

segments, info = model.transcribe(str(SAMPLE), word_timestamps=True, language="en")
words = []
text_parts = []
for seg in segments:
    text_parts.append(seg.text.strip())
    if seg.words:
        for w in seg.words:
            token = (w.word or "").strip()
            if token:
                words.append((token, w.start, w.end))

text = " ".join(text_parts).strip()
print("[path1] text:", text)
print("[path1] words:", len(words))
for token, start, end in words[:24]:
    print(f"  {start:5.2f}–{end:5.2f}  {token}")

ok = "ask" in text.lower() and len(words) >= 5
print(f"[path1] {'PASS' if ok else 'FAIL'} in {time.time() - started:.1f}s — xAI not used")
sys.exit(0 if ok else 1)
